src/components/MapPage/MapPage.jsx

```diff
@@ -1,45 +1,37 @@
-import { useState, useEffect, useMemo } from 'react';
+import { useMemo } from 'react';
 import Peripleo, { Map, Controls, ZoomControl } from '@peripleo/peripleo';
 import {
   AuthorSelect,
   ITSBStore,
   ITSBTooltip,
   ItinerariesLayer,
   IntersectionsLayer,
-  MapModeSwitch,
   MonthRangeInput,
-} from './components';
+} from '../../components';
+import { useLoaderData, useMatch } from 'react-router-dom';
+import './MapPage.css';
 
 const fetchData = (url) => fetch(url).then((res) => res.json());
 
-export function App() {
-  const [authors, setAuthors] = useState();
+export const mapLoader = async () =>
+  await Promise.all([
+    fetchData('data/authors.json'),
+    fetchData('data/places.json'),
+    fetchData('data/itineraries.json'),
+  ]);
```

Comment:
One thing I noticed here: the mapLoader is triggered on every page navigation. In practice, that may not change much, since the files should be cached by the browser. But looks like there are three requests going out whenever the user toggles between trajectories and intersections route. Perhaps there's a way to avoid this.

This, BTW, also impacts the issue on search/filter state not being persistent across page changes. (I'll write a separate comment about this below.)